python/lsst/pipe/supertask/superTask.py

```diff
@@ -64,21 +69,47 @@ class SuperTask(Task):
     ----------
     config : `pex.config.Config`, optional
         Configuration for this task (an instance of self.ConfigClass,
-        which is a task-specific subclass of lsst.pex.config.Config).
+        which is a task-specific subclass of `SuperTaskConfig`).
         If not specified then it defaults to `self.ConfigClass()`.
     log : `lsst.log.Log`, optional
         Logger instance whose name is used as a log name prefix,
         or None for no prefix. Ignored if parentTask specified, in which case
         parentTask.log's name is used as a prefix.
+    initInputs : `dict`, optional
+        A dictionary of objects needed to construct this SuperTask, with keys
+        matching the keys of the dictionary returned by
+        `getInitInputDatasetTypes` and values equivalent to what would be
+        obtained by calling `Butler.get` with those DatasetTypes and no
+        data IDs.  While it is optional for the base class, subclasses
+        are permitted to require this argument.
     """
 
     canMultiprocess = True
 
     # TODO: temporary hack, I think factory should be global
     storageClassFactory = StorageClassFactory()
 
-    def __init__(self, config=None, log=None):
-        super().__init__(config=config, log=log)
+    def __init__(self, config=None, log=None, initInputs=None):
+        super().__init__(config=config, log=log,)
+
+    def getInitOutputDatasets(self):
+        """Return persistable outputs that are available immediately after the
+        task has been constructed.
+
+        Subclasses that operate on catalogs should override this method to
+        return the schema(s) of the catalog(s) they produce.
+
+        It is not necessary to return the SuperTask's configuration or other
+        provenance information in order for it to be persisted; that is the
+        responsibility of the execution system.
+
+        Returns
+        -------
+        datasets : `dict`
+            Dictionary with keys that match those of the dict returned by
+            `getInitOutputDatasetTypes` values that can be written by calling
+            `Butler.put` with those DatasetTypes and no data IDs.
+        """
 
```

Comment:
Add explicit `return None`, otherwise it looks like something is missing here?

Reply:
Yup, definitely a mistake.  I've actually made it return an empty `dict` instead.